okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -499,6 +500,13 @@ public static boolean hasBody(Response response) {
    * content types the application is interested in.
    */
   private Request networkRequest(Request request) throws IOException {
+    if (!request.isHttps()) {
+      Authenticator authenticator = client.proxyAuthenticator();
```

Comment:
Is this mean to be using proxyAuthenticator here? Where do you do preemptive authentication with client.authenticator()?